herd/libdir/arm-models/mixed/ec.cat

```diff
@@ -40,17 +40,24 @@
 include "armv8-common.cat"
 include "sca.cat"
 
-let preorder-cb = IM0 | lob | scaob
+let preorder-cb = IM0 | local-hw-reqs | scaob
 let preorder-cb-lift = lift(MC,preorder-cb)
 
 with cb from linearisations(MC,preorder-cb-lift)
 ~empty cb
 
 let dcbl = delift(cb) & loc
-let rf-fwd = (W*R) & po-loc & dcbl^-1 \ intervening-write(po-loc)
-let rf-nfwd = (W*R) & dcbl \ intervening-write(dcbl | po-loc)
+let rf-fwd = (W*(Exp&R)) & po-loc & dcbl^-1 & ~intervening(W,po-loc)
+let rf-nfwd = (W*(Exp&R)) & dcbl & ~(intervening(W, (dcbl | po-loc)); [Exp & R]; po-loc?)
```

Comment:
Did you qualify with `Exp` these to pass some tests? I am wondering if maybe the best thing would be to qualify all of them?

Reply:
Good point. These qualifications are not necessary, since we do not intend to run these cat files with variants enabled. Both `R` and `W` in this case would always be `Exp`.

(If we were to run them with variants. We would have `rf` for `Exp&R`, `Imp&TTD&R`, `Imp&Instr&R` defined separately. `W` would have qualifications appropriate for the variant.)

Reply:
That said, leaving `W` unqualified strikes me as future-proof, since with `-variant vmsa` Explicit Reads will be potentially reading from Hardware Updates. Leaving this unresolved in case if you prefer to add Exp everywhere or drop it everywhere, @relokin . (Should not affect regressions.)

Reply:
I am mostly thinking what how we transliterate these clauses. Because the flags are not going to be transliterated. Let's leave this open. For now, your argument is very convincing and I agree with you it should cause any regressions.